### PR TITLE
Cleanup querystring params with null values to prevent reverse routing failures.

### LINKED ISSUE #14404: Pagination Helper creating wrong links when using root URL
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 4.0.5
* Platform and Target: PHP 7.3, MySQL, Ubuntu

### What you did
I added `$builder->connect('/', ['controller' => 'Pictures', 'action' => 'index']);` to my routes.php to list my pictures at the root of my url in paginated form.

### What happened
The URL generated by `$this->Paginator->first()` and ` $this->Paginator->prev()` (for the first page) is now **mywebsite.com/pictures**. When I remove `$builder->fallbacks();` from routes.php I get Missing-Route-Exception-Error.

### What you expected to happen
I'd expect the first- and prev-functions to return the right URL for the first page, which after my change in the routes.php should be **mywebsite.com** instead of **mywebsite.com/pictures**. Also it shouldn't give me the Missing-Route-Exception-Error after removing the fallback as I already declared the route in my routes.php.

### PR DESCRIPTION
**Pull request by ADmad**
Fixes #14404

### CODE REVIEW

File: tests/TestCase/View/Helper/PaginatorHelperTest.php [hunk 1]
```diff
@@ -1151,7 +1152,7 @@ public function testGenerateUrlMultiplePagination()
     public function testGenerateUrlMultiplePaginationQueryStringData()
     {
         $request = new ServerRequest([
-            'url' => '/posts/index/',
+            'url' => '/posts//',
```
Comment (dereuromark) [inline tests/TestCase/View/Helper/PaginatorHelperTest.php]: should that be one trailing slash?

**Review by ndru123 — APPROVED** [review]

**Comment by othercorey** [comment]
Do we need to back-port this?

**Comment by ADmad** [comment]
Yup the fix probably should be backported. But cherry picking the commit might cause lot of conflicts in the test file so better to directly make that one line change and update tests as required.

**Comment by ADmad** [comment]
Actually this issue doesn't exist in 3.x because there query string params are not nested under `?` key array.

**Pull request merged**